src/main/java/io/vertx/core/net/impl/transport/Transport.java

```diff
@@ -214,7 +214,9 @@ public void configure(ClientOptionsBase options, Bootstrap bootstrap) {
     setOption("TCK_FASTOPEN", options.isTcpFastOpen(), setter);
     setOption("SO_REUSEPORT", options.isReusePort(), setter);
     if (options.getLocalAddress() != null) {
-      bootstrap.localAddress(options.getLocalAddress(), 0);
+      bootstrap.localAddress(options.getLocalAddress(), options.getLocalPort());
+    } else if (options.getLocalPort() != 0) {
```

Comment:
To be consistent with the previous line you should replace `else if (options.getLocalPort() != 0) {` by `else {`